### PR TITLE
[INLONG-11494][Sort] Add Kafka connector on Flink 1.18

### PR DESCRIPTION
**Pull request by PeterZh6**
Fixes [#11494 [Feature][Sort] Add Kafka connector on Flink 1.18](https://github.com/apache/inlong/issues/11494)

### Motivation

This PR adds a Kafka connector to `inlong-sort` for Flink 1.18 support.

### Modifications

- Migrated the Kafka connector from [flink-kafka-connector 3.2.0](https://github.com/apache/flink-connector-kafka/releases/tag/3.2.0) to support Flink 1.18.
- Updated the relevant Maven `pom.xml` to reflect the changes.

### Verifying this Change

- [ ] This change is a trivial rework/code cleanup without any test coverage.
- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*
- [x] This change added tests and can be verified as follows:

  Further formal testing will be introduced in [#11495](https://github.com/apache/inlong/issues/11495).  
  Currently, a `KafkaProducer` is used to send messages to Kafka, and these messages are consumed by `sort-connectors-v1.18`, which is introduced in this PR.  
  Since there is no verified functional sink yet, the built-in `file-system` sink is used in this test.

  In the example below, we can see that the messages in Kafka are successfully extracted and written to `/tmp/output_file`.

#### Example Flink SQL:

```sql
CREATE TABLE kafka_source (
    `message` STRING
) WITH (
    'connector' = 'kafka-inlong',
    'topic' = 'test-topic',
    'properties.bootstrap.servers' = 'kafka:9092',
    'properties.group.id' = 'flink-group',
    'scan.startup.mode' = 'earliest-offset',
    'format' = 'json'
);

CREATE TABLE file_sink (
    `message` STRING
) WITH (
    'connector' = 'filesystem',
    'path' = 'file:///tmp/output_file',
    'format' = 'json'
);

INSERT INTO file_sink
SELECT * FROM kafka_source;
```

**Output:**
The messages in Kafka have been successfully extracted and transported to `/tmp/output_file`. See the screenshot below for the result:

kafka_test2_edited

### Documentation

- **Does this pull request introduce a new feature?** Yes
- **If yes, how is the feature documented?** N/A
- **If a feature is not applicable for documentation, explain why:** The version change of the sort connectors is not reflected on the official website.

### CODE REVIEW

**Comment by PeterZh6** [comment]
Sink function of this connector not tested yet.

**Comment by aloyszhang** [comment]
please add license if there are some copied codes from other project.

**Comment by PeterZh6** [comment]
> please add license if there are some copied codes from other project.

Thanks for the reminder!
All the copied codes are from [flink-connector-kafka](https://github.com/apache/flink-connector-kafka). I believe they call follow Apache 2.0 license, so I don't have to add the license once more.

**Comment by aloyszhang** [comment]
Please ref to [inlong/licenses/inlong-sort-connectors/LICENSE](https://github.com/apache/inlong/blob/master/licenses/inlong-sort-connectors/LICENSE), you should add the copied files to this file.

**Comment by PeterZh6** [comment]
> Please ref to [inlong/licenses/inlong-sort-connectors/LICENSE](https://github.com/apache/inlong/blob/master/licenses/inlong-sort-connectors/LICENSE), you should add the copied files to this file.

oh I see, sorry for the misunderstanding

**Review by aloyszhang — APPROVED** [review]

File: inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/source/KafkaSource.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+ * <p>{@link org.apache.flink.connector.kafka.source.enumerator.KafkaSourceEnumerator} only supports
+ * adding new splits and not removing splits in split discovery.
+ *
+ * <p>See {@link KafkaSourceBuilder} for more details on how to configure this source.
+ *
+ * @param <OUT> the output type of the source.
+ * copied from org.apache.flink:flink-connector-kafka:1.18.0
+ */
+// TODO: Add a variable metricSchema to report audit information
+@PublicEvolving
```
Comment (vernedeng) [inline inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/source/KafkaSource.java:89]: This flink annotation should be removed

File: inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/source/KafkaSource.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        return new KafkaSourceEnumerator(
+                subscriber,
+                startingOffsetsInitializer,
+                stoppingOffsetsInitializer,
+                props,
+                enumContext,
+                boundedness);
+    }
+
+    @Internal
```
Comment (vernedeng) [inline inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/kafka/src/main/java/org/apache/inlong/sort/kafka/source/KafkaSource.java:203]: ditto

**Review by vernedeng — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**